site/hooks/useWeb3/reducers.tsx

```diff
@@ -0,0 +1,68 @@
+import { ethers } from "ethers";
+
+export type Web3ProviderState = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  provider: any;
```

Comment:
`any` is too dangerous here since so many components will rely on it. Try `JsonRPCPRovider` from ethers, i believe that should work.

Reply:
and remove that eslint-disable once you find a valid type

Reply:
based on the initial state it can also be null ya?

Reply:
`providers.JsonRPCProvider` doesn't satisfy the types here :(

this should be typed as the return value of `web3Modal.connect()` but even web3modal package types it as `any` hahah

Reply:
🤮 

> `providers.JsonRPCProvider` doesn't satisfy the types here :(
> 
> this should be typed as the return value of `web3Modal.connect()` but even web3modal package types it as `any` hahah